src/utils/schemaEndpointResolver.js

```diff
@@ -0,0 +1,40 @@
+const memoize = require('memoizee');
+
+// This logic is wrapped into class to have isolated memoization contexts
+class SchemaEndpointResolver {
+    constructor() {
+        this.getMethodSchema = memoize(getMethodSchemaInternal);
+    }
+}
+
+function getMethodSchemaInternal(schemas, path, method) {
+    const methodLowerCase = method.toLowerCase();
+    const routePath = pathMatcher(schemas, path);
+    const route = schemas[routePath];
+
+    if (route && route[methodLowerCase] && route[methodLowerCase]) {
```

Comment:
duplicate `route[methodLowerCase]` in condition

Reply:
Fixed, thanks!